tex/generic/pgf/frontendlayer/tikz/tikz.code.tex

```diff
@@ -2131,34 +2131,34 @@
 % Central dispatcher for commands
 \def\tikz@handle{%
   \pgfutil@switch\pgfutil@ifx\pgf@let@token{%
-    {(}{\tikz@movetoabs}%)
-    {+}{\tikz@movetorel}%
-    {-}{\tikz@lineto}%
-    {.}{\tikz@dot}%
-    {r}{\tikz@rect}%
-    {n}{\tikz@fig}%
-    {[}{\tikz@parse@options}%]
-    {c}{\tikz@cchar}%
-    {\bgroup}{\tikz@beginscope}%
-    {\egroup}{\tikz@endscope}%
-    {;}{\tikz@finish}%
-    {a}{\tikz@a@char}%
-    {e}{\tikz@e@char}%
-    {g}{\tikz@g@char}%
-    {s}{\tikz@schar}%
-    {|}{\tikz@vh@lineto}%
-    {p}{\pgfsetmovetofirstplotpoint\tikz@pchar}%
-    {t}{\tikz@to}%
-    {\pgfextra}{\tikz@extra}%
-    {\foreach}{\tikz@foreach}%
-    {f}{\tikz@fchar}%
-    {\pgf@stop}{\relax}%
-    {\par}{\tikz@scan@next@command}%
-    {d}{\tikz@decoration}%
-    {l}{\tikz@l@char}%
-    {:}{\tikz@colon@char}%
-    {\relax}{\relax\tikz@scan@next@command}%
-  }{}{\tikz@resetexpandcount\tikz@expand}%
+    {(}{\let\pgfutil@next\tikz@movetoabs}%)
+    {+}{\let\pgfutil@next\tikz@movetorel}%
+    {-}{\let\pgfutil@next\tikz@lineto}%
+    {.}{\let\pgfutil@next\tikz@dot}%
+    {r}{\let\pgfutil@next\tikz@rect}%
+    {n}{\let\pgfutil@next\tikz@fig}%
+    {[}{\let\pgfutil@next\tikz@parse@options}%]
+    {c}{\let\pgfutil@next\tikz@cchar}%
+    {\bgroup}{\let\pgfutil@next\tikz@beginscope}%
+    {\egroup}{\let\pgfutil@next\tikz@endscope}%
+    {;}{\let\pgfutil@next\tikz@finish}%
+    {a}{\let\pgfutil@next\tikz@a@char}%
+    {e}{\let\pgfutil@next\tikz@e@char}%
+    {g}{\let\pgfutil@next\tikz@g@char}%
+    {s}{\let\pgfutil@next\tikz@schar}%
+    {|}{\let\pgfutil@next\tikz@vh@lineto}%
+    {p}{\def\pgfutil@next{\pgfsetmovetofirstplotpoint\tikz@pchar}}%
```

Comment:
`\pgfsetmovetofirstplotpoint` doesn't scan for an argument, so it can be inserted in-place.
```suggestion
    {p}{\pgfsetmovetofirstplotpoint\let\pgfutil@next\tikz@pchar}%
```